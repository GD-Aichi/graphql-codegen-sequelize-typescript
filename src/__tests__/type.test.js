const {
  schemaToTemplateContext,
  makeExecutableSchema
} = require('graphql-codegen-core');
const {
  compileTemplate
} = require('graphql-codegen-compiler');
const configTypescript = require('../index');

describe('Multiple Files', () => {
  const compileAndBuildContext = (typeDefs) => {
    const schema = makeExecutableSchema({
      typeDefs,
      resolvers: {},
      allowUndefinedInResolve: true
    });

    return schemaToTemplateContext(schema);
  };

  let config;

  beforeAll(() => {
    config = configTypescript.default;

  });

  describe('Schema', () => {
    it('should generate the correct types when using only simple Query', () => {
      const templateContext = compileAndBuildContext(`
        type Address {
          id: Int!
        }
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(3);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('loaders.ts');
      expect(compiled[2].filename).toMatch('loaders-export.ts');

    });

    it('should generate the correct types when directive entity', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity {
          id: Int!
        }

        directive @entity on FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(3);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('loaders.ts');
      expect(compiled[2].filename).toMatch('loaders-export.ts');
    });

    it('should generate the correct types when directive Column', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity {
          id: Int!
          name: String @Column
        }

        directive @entity on FIELD_DEFINITION
        directive @Column(dataType: String) on FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(3);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('loaders.ts');
      expect(compiled[2].filename).toMatch('loaders-export.ts');
    });

    it('should generate the correct types when directive Column with dataType', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity {
          id: Int!
          name: String @Column
          description: String @Column(dataType: "TEXT")
        }

        directive @entity on FIELD_DEFINITION
        directive @Column(dataType: String) on FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(3);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('loaders.ts');
      expect(compiled[2].filename).toMatch('loaders-export.ts');
    });
    it('should generate the correct types when directive BelongsTo', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity {
          id: Int!
          description: Description @BelongsTo
        }

        type Description {
          text: String
        }

        directive @entity on FIELD_DEFINITION
        directive @BelongsTo(foreignKey: String) on FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(4);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('description.type.ts');
      expect(compiled[1].content).toMatchSnapshot();
      expect(compiled[2].filename).toMatch('loaders.ts');
      expect(compiled[3].filename).toMatch('loaders-export.ts');
    });

    it('should generate the correct types when directive BelongsTo with foreignKey', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity {
          id: Int!
          description: Description @BelongsTo(foreignKey: "descriptionTestId")
        }

        type Description {
          text: String
        }

        directive @entity on FIELD_DEFINITION
        directive @BelongsTo(foreignKey: String) on FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(4);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('description.type.ts');
      expect(compiled[1].content).toMatchSnapshot();
      expect(compiled[2].filename).toMatch('loaders.ts');
      expect(compiled[3].filename).toMatch('loaders-export.ts');
    });

    it('should generate the correct types when directive HasMany with foreignKey', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity {
          id: Int!
          description: [Description] @HasMany(foreignKey: "addressId")
        }

        type Description {
          text: String
          addressId: Address @BelongsTo
        }

        directive @entity on FIELD_DEFINITION
        directive @BelongsTo(foreignKey: String) on FIELD_DEFINITION
        directive @HasMany(foreignKey: String!) on FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(4);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('description.type.ts');
      expect(compiled[1].content).toMatchSnapshot();
      expect(compiled[2].filename).toMatch('loaders.ts');
      expect(compiled[3].filename).toMatch('loaders-export.ts');
    });

    it('should generate the correct types when directive BelongsTo with foreignKey', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity {
          id: Int!
          description: [Description] @BelongsToMany(through: "AddressDescription")
        }

        type Description @entity {
          text: String
          Address: [Address] @BelongsToMany(through: "AddressDescription")
        }

        directive @entity on FIELD_DEFINITION
        directive @BelongsToMany(through: String!, foreignKey: String, otherKey: String) on FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(4);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('description.type.ts');
      expect(compiled[1].content).toMatchSnapshot();
      expect(compiled[2].filename).toMatch('loaders.ts');
      expect(compiled[3].filename).toMatch('loaders-export.ts');
    });

    it('should generate the correct types when directive BelongsTo with foreignKey, foreignKey and otherKey', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity {
          id: Int!
          description: [Description] @BelongsToMany(through: "AddressDescription", foreignKey: "addressTestId", otherKey:"descriptionTestId")
        }

        type Description @entity {
          text: String
          Address: [Address] @BelongsToMany(through: "AddressDescription", foreignKey: "descriptionTestId", otherKey:"addressTestId")
        }

        directive @entity on FIELD_DEFINITION
        directive @BelongsToMany(through: String!, foreignKey: String, otherKey: String) on FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(4);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('description.type.ts');
      expect(compiled[1].content).toMatchSnapshot();
      expect(compiled[2].filename).toMatch('loaders.ts');
      expect(compiled[3].filename).toMatch('loaders-export.ts');
    });

    it('should generate the correct types when directive cacheControl on object', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity @cacheControl {
          id: Int!
          name: String
        }

        directive @entity on FIELD_DEFINITION
        directive @cacheControl(maxAge: String, maxItems: Int) on OBJECT | FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(3);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('loaders.ts');
      expect(compiled[2].filename).toMatch('loaders-export.ts');
    });

    it('should generate the correct types when directive cacheControl on object with maxAge and maxItems', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity @cacheControl(maxAge: "2d", maxItems: 1000) {
          id: Int!
          name: String
        }

        directive @entity on FIELD_DEFINITION
        directive @cacheControl(maxAge: String, maxItems: Int) on OBJECT | FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(3);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('loaders.ts');
      expect(compiled[2].filename).toMatch('loaders-export.ts');
    });

    it('should generate the correct types when directive cacheControl on field', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity {
          id: Int!
          name: String @cacheControl
        }

        directive @entity on FIELD_DEFINITION
        directive @cacheControl(maxAge: String, maxItems: Int) on OBJECT | FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(3);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('loaders.ts');
      expect(compiled[2].filename).toMatch('loaders-export.ts');
    });

    it('should generate the correct types when directive cacheControl on field with maxAge and maxItems', () => {
      const templateContext = compileAndBuildContext(`
        type Address @entity {
          id: Int!
          name: String @cacheControl(maxAge: "2d", maxItems: 1000)
        }

        directive @entity on FIELD_DEFINITION
        directive @cacheControl(maxAge: String, maxItems: Int) on OBJECT | FIELD_DEFINITION
      `);
      const compiled = compileTemplate(config, templateContext);
      expect(compiled.length).toBe(3);
      expect(compiled[0].filename).toBe('address.type.ts');
      expect(compiled[0].content).toMatchSnapshot();
      expect(compiled[1].filename).toMatch('loaders.ts');
      expect(compiled[2].filename).toMatch('loaders-export.ts');
    });
  });
});
