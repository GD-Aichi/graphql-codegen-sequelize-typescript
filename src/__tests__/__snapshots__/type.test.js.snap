// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Files Schema should generate the correct types when directive BelongsTo 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';
import { DescriptionModel } from '../description.type';

import { Description } from './description.type';


export interface Address {
  id: number;
  description?: Description;
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

  @Column
  public descriptionId: number;

  @BelongsTo(() => DescriptionModel, 'descriptionId')
  public description: DescriptionModel;

}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive BelongsTo 2`] = `
"/* tslint:disable */



export interface Description {
  text?: string;
}


"
`;

exports[`Multiple Files Schema should generate the correct types when directive BelongsTo with foreignKey 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';
import { DescriptionModel } from '../description.type';

import { Description } from './description.type';


export interface Address {
  id: number;
  description?: Description;
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

  @Column
  public descriptionTestId: number;

  @BelongsTo(() => DescriptionModel, 'descriptionTestId')
  public description: DescriptionModel;

}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive BelongsTo with foreignKey 2`] = `
"/* tslint:disable */



export interface Description {
  text?: string;
}


"
`;

exports[`Multiple Files Schema should generate the correct types when directive BelongsTo with foreignKey 3`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';
import { DescriptionModel } from '../description.type';
import { AddressDescriptionModel } from '../addressDescription.type';

import { Description } from './description.type';


export interface Address {
  id: number;
  description?: Description[];
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

  @BelongsToMany(() => DescriptionModel, () => AddressDescriptionModel, 'addressId', 'descriptionId')
  public description: DescriptionModel[];
}

export class AddressDescriptionDB<T> extends IdEntityModel<T> {
  @PrimaryKey
  @Column
  public addressId: number;

  @BelongsTo(() => AddressModel, 'addressId')
  public address: AddressModel;

  @PrimaryKey
  @Column
  public descriptionId: number;

  @BelongsTo(() => DescriptionModel, 'descriptionId')
  public description: DescriptionModel;
}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive BelongsTo with foreignKey 4`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { DescriptionModel } from '../description.type';
import { AddressModel } from '../address.type';
import { AddressDescriptionModel } from '../addressDescription.type';

import { Address } from './address.type';


export interface Description {
  text?: string;
  Address?: Address[];
}


const loader: ILoaders<DataLoader<number | string, DescriptionModel>>[] = [];

export class DescriptionDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, DescriptionModel>>[] = loader;

  @BelongsToMany(() => AddressModel, () => AddressDescriptionModel, 'descriptionId', 'addressId')
  public Address: AddressModel[];
}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive BelongsTo with foreignKey, foreignKey and otherKey 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';
import { DescriptionModel } from '../description.type';
import { AddressDescriptionModel } from '../addressDescription.type';

import { Description } from './description.type';


export interface Address {
  id: number;
  description?: Description[];
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

  @BelongsToMany(() => DescriptionModel, () => AddressDescriptionModel, 'addressTestId', 'descriptionTestId')
  public description: DescriptionModel[];
}

export class AddressDescriptionDB<T> extends IdEntityModel<T> {
  @PrimaryKey
  @Column
  public addressTestId: number;

  @BelongsTo(() => AddressModel, 'addressTestId')
  public address: AddressModel;

  @PrimaryKey
  @Column
  public descriptionTestId: number;

  @BelongsTo(() => DescriptionModel, 'descriptionTestId')
  public description: DescriptionModel;
}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive BelongsTo with foreignKey, foreignKey and otherKey 2`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { DescriptionModel } from '../description.type';
import { AddressModel } from '../address.type';
import { AddressDescriptionModel } from '../addressDescription.type';

import { Address } from './address.type';


export interface Description {
  text?: string;
  Address?: Address[];
}


const loader: ILoaders<DataLoader<number | string, DescriptionModel>>[] = [];

export class DescriptionDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, DescriptionModel>>[] = loader;

  @BelongsToMany(() => AddressModel, () => AddressDescriptionModel, 'descriptionTestId', 'addressTestId')
  public Address: AddressModel[];
}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive Column 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';



export interface Address {
  id: number;
  name?: string;
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

  @Column
  public name: string;

}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive Column with dataType 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';



export interface Address {
  id: number;
  name?: string;
  description?: string;
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

  @Column
  public name: string;

  @Column(DataType.TEXT)
  public description: string;

}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive HasMany with foreignKey 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';
import { DescriptionModel } from '../description.type';

import { Description } from './description.type';


export interface Address {
  id: number;
  description?: Description[];
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

  @BelongsTo(() => DescriptionModel, 'addressId')
  public description: DescriptionModel;

}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive HasMany with foreignKey 2`] = `
"/* tslint:disable */

import { Address } from './address.type';


export interface Description {
  text?: string;
  addressId?: Address;
}


"
`;

exports[`Multiple Files Schema should generate the correct types when directive cacheControl on field 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';

import { addressBynameLoader } from './loaders';


export interface Address {
  id: number;
  name?: string;
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];
loader.push({ 'name': addressBynameLoader });

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive cacheControl on field with maxAge and maxItems 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';

import { addressBynameLoader } from './loaders';


export interface Address {
  id: number;
  name?: string;
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];
loader.push({ 'name': addressBynameLoader });

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive cacheControl on object 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';

import { addressByIdLoader } from './loaders';


export interface Address {
  id: number;
  name?: string;
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];
loader.push({ id: addressByIdLoader });

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive cacheControl on object with maxAge and maxItems 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';

import { addressByIdLoader } from './loaders';


export interface Address {
  id: number;
  name?: string;
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];
loader.push({ id: addressByIdLoader });

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

}
"
`;

exports[`Multiple Files Schema should generate the correct types when directive entity 1`] = `
"/* tslint:disable */
import {Model, BelongsTo, Column, IsUUID, PrimaryKey, HasMany, BelongsToMany, DataType} from 'sequelize-typescript';
import {IdEntityModel, ILoaders} from '../core/IdEntityModel';
import * as DataLoader from 'dataloader';
import { AddressModel } from '../address.type';



export interface Address {
  id: number;
}


const loader: ILoaders<DataLoader<number | string, AddressModel>>[] = [];

export class AddressDB<T extends Model<T>> extends IdEntityModel<T> {

  static loader: ILoaders<DataLoader<number | string, AddressModel>>[] = loader;

}
"
`;

exports[`Multiple Files Schema should generate the correct types when using only simple Query 1`] = `
"/* tslint:disable */



export interface Address {
  id: number;
}


"
`;
